dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -194,6 +196,14 @@ private PermissionStates getPermissionStates(Collection<String> pendingPermissio
    * https://github.com/Karumi/Dexter/issues/86
    */
   private int checkSelfPermission(Activity activity, String permission) {
+    // If we want to force permission, return Permission denied here, so that permission is asked later. Otherwise,
+    // system may answer PERMISSION_GRANTED and we will not ask for permission explicitly.
+    // This is working because this method is only called by
+    // DexterInstance.onActivityReady() -> DexterInstance.getPermissionStates(), and then all PERMISSION_DENIED tied
+    // with a permission string are handled in asking for permission explicitly.
+    if (forcePermissionRequest) {
+      return PackageManager.PERMISSION_DENIED;
```

Comment:
Semantically, this is broken. The permission `permission` which is being checked at this moment, is not denied, could be granted. What you want to do is bypass the system check. So what I would do is set that login in `getPermissionStates` and there you choose what to do, if force it/them or not.